### PR TITLE
+ Added some additional hooks

### PR DESCRIPTION
**Pull request by jdarwood007**
Some hooks that would be very handy to add into SMF.

### CODE REVIEW

**Comment by jdarwood007** [comment]
I should note I did little testing with these.  But they seem solid.

I also realize I didn't sign off, using a new Git GUI tool.  I tried via command line to amend the message but it complains about a fast forward error.  Can we just ignore that so I don't have to try and figure out how to add that.

File: Sources/Display.php [hunk 1]
```diff
@@ -1479,23 +1479,31 @@ function Download()
 
 	$_REQUEST['attach'] = isset($_REQUEST['attach']) ? (int) $_REQUEST['attach'] : (int) $_REQUEST['id'];
 
-	// This checks only the current board for $board/$topic's permissions.
-	isAllowedTo('view_attachments');
+	// Do we have a hook wanting to use our attachment system? We use $attachRequest to prevent accidental usage of $request.
+	$attachRequest = null;
+    call_integration_hook('integrate_download_request', array(&$attachRequest));
```
Comment (XinYenFon) [inline Sources/Display.php]: My OCD :P

**Comment by Oldiesmann** [comment]
Besides the sign off thing, the only issue I see is that the indentation is inconsistent in several areas.

I'm not sure if we can ignore the sign off requirement. I did last night, but only because I was tired and wasn't paying attention...

**Comment by jdarwood007** [comment]
Fixed those up and signed off.

File: Sources/ManageAttachments.php [hunk 2]
```diff
@@ -1530,6 +1537,8 @@ function RepairAttachments()
 		for (; $_GET['substep'] < $thumbnails; $_GET['substep'] += 500)
 		{
 			$to_remove = array();
+			$ignore_ids = array(0);
+			call_integration_hook('integrate_repair_attachments_nomsg', array(&$ignore_ids, $_GET['substep'], $_GET['substep'] += 500));
```
Comment (jdarwood007) [inline Sources/ManageAttachments.php:1541]: This is properly tabbed, just displays weird in GitHub.

**Pull request merged**